RecoLocalCalo/HcalRecProducers/plugins/alpaka/Mahi.dev.cc

```diff
@@ -0,0 +1,1470 @@
+                    sample,
+                    noiseADC,
+                    sample,
+                    noisePhotoSq,
+                    outputGPU.chi2()[gch]);
+#endif
+
+                // store to global memory
+                amplitudesForChannel[sampleWithinWindow] = amplitude;
+                noiseTermsForChannel[sampleWithinWindow] = noiseTerm;
+                electronicNoiseTermsForChannel[sampleWithinWindow] = pedestalWidth;
+
+              }  // loop for sample
+            }    // loop for channels
+          }      // loop for channgel groups
+        }
+      };  //Kernel_prep1d_sameNumberOfSamples
+      class Kernel_prep_pulseMatrices_sameNumberOfSamples {
+      public:
+        template <typename TAcc, typename = std::enable_if_t<alpaka::isAccelerator<TAcc>>>
+        ALPAKA_FN_ACC void operator()(TAcc const& acc,
+                                      float* pulseMatrices,
+                                      float* pulseMatricesM,
+                                      float* pulseMatricesP,
+                                      HcalMahiPulseOffsetsPortableDevice::ConstView pulseOffsets,
+                                      float const* amplitudes,
+                                      IProductTypef01::ConstView f01HEDigis,
+                                      IProductTypef5::ConstView f5HBDigis,
+                                      IProductTypef3::ConstView f3HBDigis,
+                                      int8_t* soiSamples,
+                                      HcalMahiConditionsPortableDevice::ConstView mahi,
+                                      HcalRecoParamWithPulseShapeDevice::ConstView recoParamsWithPS,
+                                      float const meanTime,
+                                      float const timeSigmaSiPM,
+                                      float const timeSigmaHPD,
+                                      bool const applyTimeSlew,
+                                      float const tzeroTimeSlew,
+                                      float const slopeTimeSlew,
+                                      float const tmaxTimeSlew) const {
+          //2nd index = pulse
+          auto const npulses(alpaka::getWorkDiv<alpaka::Block, alpaka::Elems>(acc)[1u]);
+          //3rd index = sample
+          auto const nsamples(alpaka::getWorkDiv<alpaka::Block, alpaka::Elems>(acc)[2u]);
+
+          auto const nchannels = f01HEDigis.size() + f5HBDigis.size() + f3HBDigis.size();
+          auto const nchannelsf015 = f01HEDigis.size() + f5HBDigis.size();
+
+          auto const recoParamView = recoParamsWithPS.recoParamView();
+          auto const pulseShapeView = recoParamsWithPS.pulseShapeView();
+
+          //Loop over all groups of channels
+          for (auto group : uniform_groups_z(acc, nchannels)) {
+            //Loop over each channel
+            for (auto channel : uniform_group_elements_z(acc, group, nchannels)) {
+              auto const gch = channel.global;
```

Comment:
Could this be more simply
```suggestion
          for (auto channel : uniform_elements_z(acc, nchannels)) {
```
using `channel` instead of `gch` in the rest of the loop ?

Reply:
Done 4bcedbfa3762f46753bcb6564102898708f5562b